src/main/1.3/scala/com/holdenkarau/spark/testing/JavaStreamingSuitebase.scala

```diff
@@ -85,65 +61,129 @@ class JavaStreamingSuiteBase extends JavaSuiteBase with StreamingSuiteCommon {
     // Match the output with the expected output
     assertEquals("Number of outputs do not match", expectedOutput.size, output.size)
     for (i <- output.indices) {
-      if (useSet) {
-        implicit val config = Bag.configuration.compact[V]
-        compareArrays[V](Bag(expectedOutput(i): _*).toArray,
-          Bag(output(i): _*).toArray)
-      } else {
+      if (ordered) {
         compareArrays[V](expectedOutput(i).toArray, output(i).toArray)
+      } else {
+        implicit val config = Bag.configuration.compact[V]
+        compareArrays[V](Bag(expectedOutput(i): _*).toArray, Bag(output(i): _*).toArray)
       }
     }
+
     logInfo("Output verified successfully")
   }
 
   /**
-   * Test unary DStream operation with a list of inputs, with number of
-   * batches to run same as the number of expected output values
-   */
-  def testOperation[U, V](
-    input: JList[JList[U]],
-    operation: JFunction[JavaDStream[U], JavaDStream[V]],
-    expectedOutput: JList[JList[V]]) {
-    testOperation[U, V](input, operation, expectedOutput, -1, false)
+    * Test unary DStream operation with a list of inputs, with number of
+    * batches to run same as the number of input values.
+    * You can simulate the input batch as a List of values or as null to simulate empty batch.
+    *
+    * @param input Sequence of input collections
+    * @param operation Binary DStream operation to be applied to the 2 inputs
+    * @param expectedOutput Sequence of expected output collections
+    */
+  def testOperation[U, V](input: JList[JList[U]],
+                          operation: JFunction[JavaDStream[U], JavaDStream[V]],
+                          expectedOutput: JList[JList[V]]) {
+    testOperation[U, V](input, operation, expectedOutput, false)
```

Comment:
It's default value in `StreamingSuiteBase` is false. I like to keep both Scala & Java the same default value.